autogpt_platform/backend/backend/notifications/email.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import pathlib
+
+from backend.util.text import TextFormatter
+from postmarker.core import PostmarkClient
+from postmarker.models.emails import EmailManager
+from prisma.enums import NotificationType
+
+from backend.data.notifications import (
+    NotificationEventModel,
+    NotificationTypeOverride,
+    T_co,
+)
+from backend.util.settings import Settings
+
+logger = logging.getLogger(__name__)
+settings = Settings()
+
+
+# The following is a workaround to get the type checker to recognize the EmailManager type
+# This is a temporary solution and should be removed once the Postmark library is updated
+# to support type annotations.
+class TypedPostmarkClient(PostmarkClient):
+    emails: EmailManager
+
+
+class EmailSender:
+    def __init__(self):
+        self.postmark = TypedPostmarkClient(
+            server_token=settings.secrets.postmark_server_api_token
+        )
+        self.formatter = TextFormatter()
+
+    def send_templated(
+        self,
+        notification: NotificationType,
+        user_email: str,
+        data: NotificationEventModel[T_co] | list[NotificationEventModel[T_co]],
+    ):
+        body = self._get_template(notification)
+        # use the jinja2 library to render the template
+        body = self.formatter.format_string(body, data)
+        logger.info(
+            f"Sending email to {user_email} with subject {"subject"} and body {body}"
+        )
+        self._send_email(user_email, "subject", body)
+
+    def _get_template(self, notification: NotificationType):
+        # convert the notification type to a notification type override
+        notification_type_override = NotificationTypeOverride(notification)
```

Comment:
this override is a superset of the Notification Type that has useful functions built in